Add LoginTests and RegisterTests

plexdeath opened the pull request:
Add PageObjects
Add Utils
Add BaseTest and DataTests

Mosdig committed:
Add LoginTests and RegisterTests
Add PageObjects
Add Utils
Add BaseTest and DataTests

Mosdig committed:
Add GoToTests add new many Tests
add Many methods add RecoveryPage.java
This is Final version
This version starts Chrome and Yandex Browser's
Chrome bug assertion feelings

mkouznetsov-git commented on pom.xml:
```diff
@@ -0,0 +1,122 @@
+            <version>4.29.0</version>
+        </dependency>
+        <dependency>
+            <groupId>io.qameta.allure</groupId>
```
Дубликат. Такая зависимость уже объявлена выше.

plexdeath replied:
Лишние зависимости удалил , это я экспериментировал с отчетами и идея их задублировала , спасибо что заметил

mkouznetsov-git commented on pom.xml:
```diff
@@ -0,0 +1,122 @@
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.qameta.allure</groupId>
```
Дубликат. Такая зависимость уже объявлена выше.

plexdeath replied:
Лишние зависимости удалил , это я экспериментировал с отчетами и идея их задублировала , спасибо что заметил

mkouznetsov-git commented on src/test/java/GoToTests.java:
```diff
@@ -0,0 +1,112 @@
+    @Test
+    @DisplayName("Проверка перехода по клику на «Конструктор» и на логотип Stellar Burgers.")
+    @Description("Проверить переход по клику на «Конструктор» и на логотип Stellar Burgers.")
+    public void testConstructorLogoVerify() throws InterruptedException {
```
В этом тесте две разных проверки. Это некорректно -- иметь их в одном тестовом методе. Проверки должны быть независимыми друг от друга. Если у тебя тест на первой проверке свалится, то до второй дело не дойдёт, и она не будет выполнена. Раздели по отдельным тестовым методам свои проверки по принципу "одна проверка -- один метод".

plexdeath replied:
Вместо одного тестового метода testConstructorLogoVerify
Сделал две, чтобы соблюсти правило 1 тест 1 проверка.
testConstructorButtonRedirectsToMainPage 
testLogoClickRedirectsToMainPage
Дополнительно:
Разнес проверки с разными выводами в случае провала 
verifyMainPageUrlConstructor Ссылка ведёт не на главную страницу после клика на кнопку конструктор
verifyMainPageUrlBurger Ссылка ведёт не на главную страницу после клика на логотип

mkouznetsov-git commented on src/test/java/GoToTests.java:
```diff
@@ -0,0 +1,112 @@
+    @Test
+    @DisplayName("Проверка выхода из личного кабинета")
+    @Description("Проверить что вышли из личного кабинета")
+    public void testExitLK() throws InterruptedException {
```
Здесь то же самое.

plexdeath replied:
В данном методе testExitLK() одна проверка  
loginPage.verifyExitLkURL();
//здесь мы убеждаемся что находимся на странице с логином после выхода. 
Остальное переходы.  
Единственное что упразднил в этом методе чтобы он не главную переходил а сразу на страницу авторизации . 
Сокращаем шаги и убыстряем тест без лишнего клика. 
А вот тестовый метод verifyBunSouse. 
Разбил на 3 отдельных теста 
testSaucesTabIsActive , 
testFillingsTabIsActive, 
testBunsTabIsActive чтобы соблюсти правило 1 тест 1 проверка.

mkouznetsov-git commented on src/test/java/LoginTests.java:
```diff
@@ -0,0 +1,118 @@
+        loginPage.inputPassword(mainPage.getCreatedUserPassword());
+        loginPage.enterButtonClick();
+        mainPage.clickEnterLK();//перешли в лк c главной страницы
+        Assert.assertTrue(
```
Здесь уместнее использовать assertEquals()

plexdeath replied:
Поправил assertTrue на assertEquals во всех методах проверки
Assert.assertEquals(
    "Проверяем, что созданный пользователь отображается в интерфейсе после авторизации",
    mainPage.getCreatedUserEmail(),
    lkPage.getLkLoginText()
);

mkouznetsov-git commented on src/test/java/LoginTests.java:
```diff
@@ -0,0 +1,118 @@
+        loginPage.inputPassword(mainPage.getCreatedUserPassword());
+        loginPage.enterButtonClick();
+        mainPage.clickEnterLK();//перешли в лк c главной страницы
+        Assert.assertTrue(
```
И здесь тоже. И далее -- во всех тестах.

plexdeath replied:
Поправил assertTrue на assertEquals во всех методах проверки
Assert.assertEquals(
    "Проверяем, что созданный пользователь отображается в интерфейсе после авторизации",
    mainPage.getCreatedUserEmail(),
    lkPage.getLkLoginText()
);

mkouznetsov-git commented on src/test/java/RegisterTests.java:
```diff
@@ -0,0 +1,77 @@
+        registerPage.inputEmail(user.getEmail());
+        registerPage.inputPassword(password);
+        registerPage.clickEmail();
+        Assert.assertTrue(
```
assertEquals()

plexdeath replied:
Поправил assertTrue на assertEquals во всех методах проверки
Assert.assertEquals(
    "Проверяем, что созданный пользователь отображается в интерфейсе после авторизации",
    mainPage.getCreatedUserEmail(),
    lkPage.getLkLoginText()
);

Mosdig committed:
Add openAutorizeLkUrl
Remove testConstructorLogoVerify and verifyBunSouse
ADD testConstructorButtonRedirectsToMainPage , testLogoClickRedirectsToMainPage , testSaucesTabIsActive, testFillingsTabIsActive , testBunsTabIsActive
Change assertTrue on assertEquals
This is the final version after the edits

Mosdig committed:
This is last Final version !
in some methods, I corrected assertTrue to assertEquals.

plexdeath commented:
https://github.com/plexdeath/QA-java-diplom-3_Safonov_Vladimir/pull/1/commits/5cc66cf019a2b0715bd5776dc20561e72baaff76 
Это последний коммит не учел в прошлом замену assertTrue на assertEquals
